ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/vllm/426__model_serving_vllm_metrics.robot

```diff
@@ -0,0 +1,119 @@
+Suite Teardown    Suite Teardown
+Test Tags         KServe
+
+
+*** Variables ***
+${VLLM_RESOURCES_DIRPATH}=    ods_ci/tests/Resources/Files/llm/vllm
+${DL_POD_FILEPATH}=           ${VLLM_RESOURCES_DIRPATH}/download_model.yaml
+${SR_FILEPATH}=               ${VLLM_RESOURCES_DIRPATH}/vllm_servingruntime.yaml
+${IS_FILEPATH}=               ${VLLM_RESOURCES_DIRPATH}/vllm-gpt2_inferenceservice.yaml
+${INFERENCE_INPUT}=           @${VLLM_RESOURCES_DIRPATH}/query.json
+${INFERENCE_URL}=             http://localhost:8080/v1/chat/completions
+${METRICS_URL}=               http://localhost:8080/metrics/
+${TEST_NS}=                   vllm-gpt2
+@{SEARCH_METRICS}=            vllm:cache_config_info
+...                           vllm:num_requests_running
+...                           vllm:num_requests_swapped
+...                           vllm:num_requests_waiting
+...                           vllm:gpu_cache_usage_perc
+...                           vllm:cpu_cache_usage_perc
+...                           vllm:prompt_tokens_total
+...                           vllm:generation_tokens_total
+...                           vllm:time_to_first_token_seconds_bucket
+...                           vllm:time_to_first_token_seconds_count
+...                           vllm:time_to_first_token_seconds_sum
+...                           vllm:time_per_output_token_seconds_bucket
+...                           vllm:time_per_output_token_seconds_count
+...                           vllm:time_per_output_token_seconds_sum
+...                           vllm:e2e_request_latency_seconds_bucket
+...                           vllm:e2e_request_latency_seconds_count
+...                           vllm:e2e_request_latency_seconds_sum
+...                           vllm:avg_prompt_throughput_toks_per_s
+...                           vllm:avg_generation_throughput_toks_per_s
+
+
+*** Test Cases ***
+Verify User Can Deploy A Model With Vllm Via CLI
+    [Documentation]    Deploy a model (gpt2) using the vllm runtime and confirm that it's running
+    [Tags]    Tier1    Sanity    Resources-GPU    ODS-XXX
+    ${rc}    ${out}=    Run And Return Rc And Output    oc apply -f ${DL_POD_FILEPATH}
+    Should Be Equal As Integers    ${rc}    ${0}
+    Wait For Pods To Succeed    label_selector=gpt-download-pod=true    namespace=${TEST_NS}
+    ${rc}    ${out}=    Run And Return Rc And Output    oc apply -f ${SR_FILEPATH}
+    Should Be Equal As Integers    ${rc}    ${0}
+    ${rc}    ${out}=    Run And Return Rc And Output    oc apply -f ${IS_FILEPATH}
+    Should Be Equal As Integers    ${rc}    ${0}
+    Wait For Pods To Be Ready    label_selector=serving.kserve.io/inferenceservice=vllm-gpt2-openai
+    ...    namespace=${TEST_NS}
+    ${pod_name}=  Get Pod Name    namespace=${TEST_NS}
+    ...    label_selector=serving.kserve.io/inferenceservice=vllm-gpt2-openai
+    Start Port-forwarding    namespace=${TEST_NS}    pod_name=${pod_name}    local_port=8080   remote_port=8080
```

Comment:
Just curious as to why port forward instead of using the isvc url?

Reply:
This should not be used for serverless deployment. We should use isvc url

Reply:
FWIW I followed the information found in https://github.com/opendatahub-io/odh-model-controller/pull/200 and RHOAIENG-6264